### ISSUE TITLE
Preserve 64bit EOS filename in install_image

### ISSUE THREAD

**Issue by siguroot** [arista-eosplus/ztpserver#404]
It would be very useful to me to have the EOS image name preserved when install_image copies firmware to flash.

Currently I have this in my definition file:

```
  -
    action: install_image
    always_execute: true
    attributes:
      url: files/images/EOS64-4.30.6M.swi
      version: 4.30.6M
    name: "Install 4.30.6M"
```

When I check the flash for the resulting filename it is `EOS-4.30.6M.swi`.  I believe this is being caused by:

```
image = "EOS-{}.swi".format(version)
```

Wouldn't it be better just to carry over the filename defined in the definition?

Let me know if you want me to put in a pull request.

**Comment by Trailingslashes** [comment]
I think this might be a limitation of EOS.

**Comment by siguroot** [comment]
I'm referring to the filename on flash

**Comment by Trailingslashes** [comment]
oh that's right I misunderstood. You want EOS64 on the flash dir.

**Comment by siguroot** [comment]
I think it is potentially dangerous (at least in my environment).  If someone not aware of this issue goes to install new firmware they might install 32bit firmware as it matches the firmware filename on the arista download site.  

Perhaps a attribute (perhaps: `preserve: True`) could be added to tell `node.retrieve_url` to just use the filename in the url as the destination filename? 

Currently I'm just doing this, but surely the there is a more elegant solution:

```
    if re.search(r'EOS64', url):
        image = "EOS64-{}.swi".format(version)
    else:
        image = "EOS-{}.swi".format(version)
```

**Comment by Trailingslashes** [comment]
Do you know if its possible to verify the arch of an swi on disk?

**Comment by siguroot** [comment]
Good question, perhaps I'll create a tac case to ask this.  Thus far I have just been confirming with md5 checksum.

**Comment by siguroot** [comment]
Arista got back to me, apparently doing a checksum is the best way to verify a if a filename matches the desired architecture.  As for the 'Architecture' in the `show version` output.  That represents the architecture of the currently loaded firmware.

**Comment by dlobato** [comment]
First of all @Trailingslashes  and @siguroot thanks for the contribution.

In order to keep backward compatibility, I'd rather add a new optional attribute "image_name_from_url: True" or something like that to signal that we want to use the name provided in the url. Then if provided, we do as @siguroot implemented in #406

Any issue with this approach?

**Comment by siguroot** [comment]
Yes, this approach will work great for my environment!  Thanks

**Comment by siguroot** [comment]
I have pushed a commit to #406 that handles the option attribute "image_name_from_url"

**Comment by dlobato** [comment]
Solved with #406
